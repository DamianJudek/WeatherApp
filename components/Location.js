import React from 'react';
import { View, StyleSheet, Image, TouchableHighlight } from 'react-native';
import ASBox from './AnimatedSearchBox';

const styles = StyleSheet.create({
  container: {
    height: 50,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
    backgroundColor: 'rgba(50, 50, 50, 0.3)',
  },
  icon: {
    height: 20,
    width: 20,
  },
  iconContainer: {
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 5,
    backgroundColor: 'transparent',
  },
  baseText: {
    fontFamily: 'Roboto',
    color: '#fff',
    fontSize: 22,
  },
  heading: {
    fontSize: 35,
  },
});

class Location extends React.Component {
  render() {
    return (
      <View style={styles.container}>
        <ASBox handleCityInput={this.props.handleCityInput} />
        <TouchableHighlight
          onPress={this.props.getLocation}
          underlayColor="rgba(95, 158, 160, 0.3)"
          style={styles.iconContainer}>
          <Image
            style={styles.icon}
            resizeMode="contain"
            source={require('../assets/icons/gps.png')}
          />
        </TouchableHighlight>
      </View>
    );
  }
}

export default Location;
