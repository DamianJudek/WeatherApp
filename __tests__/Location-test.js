import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Location from '../components/Location';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer.create(<Location />).toJSON();
  expect(snap).toMatchSnapshot();
});
