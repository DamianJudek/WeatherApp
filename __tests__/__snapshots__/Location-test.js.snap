// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(50, 50, 50, 0.3)",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "flex-end",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/icons/search.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(100, 100, 100, 0.3)",
        "flex": 0,
        "height": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      onSubmitEditing={[Function]}
      placeholder=""
      placeholderTextColor="#fff"
      rejectResponderTermination={true}
      style={
        Object {
          "padding": 0,
          "width": 0,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "height": "100%",
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingVertical": 5,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/icons/gps.png",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
